### PR TITLE
clientv3/integration: close active connection to get ErrClientConnClosing

### PR DESCRIPTION
**Pull request by gyuho**
because clientv3.Close won't trigger it any more

clientv3.Close just closes watch client
instead of closing grpc connection

/cc @heyitsanthony @xiang90

### CODE REVIEW

File: clientv3/integration/watch_test.go [hunk 1]
```diff
@@ -619,9 +619,12 @@ func TestWatchErrConnClosed(t *testing.T) {
 	donec := make(chan struct{})
 	go func() {
 		defer close(donec)
-		wc.Watch(context.TODO(), "foo")
-		if err := wc.Close(); err != nil && err != grpc.ErrClientConnClosing {
-			t.Fatalf("expected %v, got %v", grpc.ErrClientConnClosing, err)
+		ch := wc.Watch(context.TODO(), "foo")
+		wc.Close()
```
Comment (heyitsanthony) [inline clientv3/integration/watch_test.go]: why?

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Thanks!

**Comment by heyitsanthony** [comment]
lgtm after adding the `defer cli.Close()`. Thanks!

**Comment by xiang90** [comment]
@gyuho Will this fix https://github.com/coreos/etcd/issues/6630?

**Comment by gyuho** [comment]
@xiang90 @heyitsanthony Fixed. PTAL.

Regarding https://github.com/coreos/etcd/pull/6768

This change would resolve that issue in that we now correctly close the gRPC connections.

I have been trying to reproduce that issue injecting some sleeps

```go
// Close shuts down the client's etcd connections.
func (c *Client) Close() error {
	c.cancel()
	time.Sleep(2 * time.Second)
	c.Watcher.Close()
	return toErr(c.ctx, c.conn.Close())
}
```

but I think we can close it for now? I tried over 5k rounds but cannot reproduce...

Thanks!

**Comment by heyitsanthony** [comment]
lgtm. No strong opinion about #6630 and this; can close/reopen if it shows up again.

**Pull request merged**